### PR TITLE
Update Specs to Include Gas Price changes

### PR DESCRIPTION
**Pull request by MitchTurner**
https://github.com/FuelLabs/fuel-specs/pull/546

This includes the changes for:
- Remove`gas_price` from tx
- Include `gas_price` on `Mint` tx
- Remove `GasPrice` policy
- Add `Tip` policy
- Make `MaxFee` mandatory

Since there is no concept of the client API in these specs, and also the implementation of tx checks, I didn't include:
- Anything wrt to `gas_price` estimate or latest `gas_price` API endpoints
- Anything wrt the metadata changes for `Checked<Tx>`

That's fine if we don't want the scope of the VM to include those sort of things, but it's not clear to me _what_ these specs are constrained by.

### CODE REVIEW

File: src/tx-format/policy.md [hunk 1]
```diff
@@ -14,11 +14,11 @@ enum PolicyType : uint32 {
...
 
-## `GasPrice`
+## `Tip`
 
-| name       | type     | description               |
-|------------|----------|---------------------------|
-| `gasPrice` | `uint64` | Gas price for transaction |
+| name       | type     | description                                                                                   |
+|------------|----------|-----------------------------------------------------------------------------------------------|
+| `gasPrice` | `uint64` | Additional, optional fee in `BASE_ASSET` to incentivize block producer to include transaction |
```
Comment (xgreenx) [inline src/tx-format/policy.md:21]:
```suggestion
| `tip` | `uint64` | Additional, optional fee in `BASE_ASSET` to incentivize block producer to include transaction |
```

File: src/protocol/tx-validity.md [hunk 2]
```diff
@@ -212,14 +212,14 @@ def max_gas(tx) -> int:
     if tx.type == TransactionType.Script:
        gas = gas + tx.gasLimit
     return gas
-
-
-def reserved_feeBalance(tx, assetId) -> int:
```
Comment (MitchTurner) [inline src/protocol/tx-validity.md:217]: I couldn't find any function with this name in our code, and it sounds like `max_fee`, so I changed it. Might be wrong.

**Review by Dentosal — APPROVED** [review]

File: src/protocol/tx-validity.md (hunk 2 above)
Comment (xgreenx) [reply src/protocol/tx-validity.md:217]: Yeah, in the code analog is `max_fee` function=) Could you also update other places here(`reserved_fee_balance`), please?

File: src/tx-format/policy.md [hunk 3]
```diff
@@ -44,11 +44,12 @@ Transaction is invalid if:
...
-| name      | type     | description                                                     |
-|-----------|----------|-----------------------------------------------------------------|
-| `max_fee` | `uint64` | The maximum fee payable by this transaction using `BASE_ASSET`. |
+| name      | type     | description                                                                                                                                |
+|-----------|----------|--------------------------------------------------------------------------------------------------------------------------------------------|
+| `max_fee` | `uint64` | The maximum fee payable by this transaction using `BASE_ASSET`. This is used to check transactions before the actual `gas_price` is known. |
 
 Transaction is invalid if:
 
+- `max_fee` is not set
```
Comment (xgreenx) [inline src/tx-format/policy.md:53]: Maybe it is better to move this check to where the gas price was.

**Review by xgreenx — CHANGES_REQUESTED** [review]
You also need to update `unavailable_balance` function to use `max_fee` policy instead of the `maxFee`(`reserved_fee_balance`) function

File: src/tx-format/policy.md (hunk 3 above)
Comment (MitchTurner) [reply src/tx-format/policy.md:53]: I'm not sure I understand. Move this section to where the `## GasPrice` section was? Which "gas price" are you referring to?

**Pull request merged**